cdc/processor/pipeline/sorter.go

```diff
@@ -102,6 +114,45 @@ func (n *sorterNode) Init(ctx pipeline.NodeContext) error {
 				if msg == nil {
 					continue
 				}
+				if msg.RawKV != nil && msg.RawKV.OpType != model.OpTypeResolved {
+					size := uint64(msg.RawKV.ApproximateSize() * 2)
+					commitTs := msg.CRTs
+
+					if time.Since(lastSendResolvedTsTime) > time.Millisecond*200 {
+						if lastCRTs > lastSentResolvedTs && commitTs > lastCRTs {
+							lastSentResolvedTs = lastCRTs
+							lastSendResolvedTsTime = time.Now()
+							ctx.SendToNextNode(pipeline.PolymorphicEventMessage(model.NewResolvedPolymorphicEvent(0, lastCRTs)))
+						}
+					}
+					err := n.flowController.Consume(commitTs, size, func() error {
+						if lastCRTs > lastSentResolvedTs {
+							lastSentResolvedTs = lastCRTs
+							lastSendResolvedTsTime = time.Now()
+							ctx.SendToNextNode(pipeline.PolymorphicEventMessage(model.NewResolvedPolymorphicEvent(0, lastCRTs)))
+						}
+						return nil
+					})
+					if err != nil {
+						if cerror.ErrFlowControllerAborted.Equal(err) {
+							log.Info("flow control cancelled for table",
+								zap.Int64("tableID", n.tableID),
+								zap.String("tableName", n.tableName))
+						} else {
+							ctx.Throw(err)
+						}
+						return nil
+					}
+					lastCRTs = commitTs
+				}
+
+				if msg.RawKV != nil && msg.RawKV.OpType == model.OpTypeResolved {
```

Comment:
We can count it, but since the quota is not strictly obeyed in this implementation, counting it is not a priority. We cannot strictly obey to quota because blocking in the middle of a Txn can deadlock the whole pipeline.